### ISSUE TITLE
Issues with MacOS build

### ISSUE THREAD

**Issue by ericraymond** [SingularSound/openbbm#34]
_These issues might be related to installing QMake version 3.1 ... using Qt version 5.14.2._

1) The readme mention using `qmake && make clean && make all`.   Should this be  `qmake -makefile bbmanager.pro  && make clean && make all`?

2) Running `qmake -makefile bbmanager.pro  && make clean && make all` results in the following errors:

```make[1]: *** No rule to make target `mac_deployment/Bundle_src/Content/Resources/Application_Icon.icns', needed by `BBManagerLean.app/Contents/Resources/Application_Icon.icns'.  Stop.
make: *** [sub-BBManagerLean-all] Error 2 ```

3) Running with beatbuddy-manager-lite.pro resukts in a lot of errors 

4) Readme should describe the difference between bbmanager.pro and beatbuddy-manager-lite.pro

**Comment by JulissaDantes** [comment]
Regarding points 1 & 2, please try again with the Qt Creator Build Settings "Shadow" option  turned off.
3. To run the application you need to use  **bbmanager.pro** instead of beatbuddy-manager-lite.pro
4. Thanks for the observation, we will be updating the README so is more clear on the subject.

**Comment by JulissaDantes** [comment]
Please check the PR #36 and confirm if the readme is clear enough regarding the point 3 and 4

**Comment by ericraymond** [comment]
> Regarding points 1 & 2, please try again with the Qt Creator Build Settings "Shadow" option turned off.

Not sure how I would do that.  I did try the suggestion in https://stackoverflow.com/questions/5171896/manually-configuring-shadow-build-in-qmake but that did not work and gave the same error:

```
mkdir my_shadow_build; cd my_shadow_build
 qmake -makefile ../bbmanager.pro && make clean && make all
```

**Comment by ericraymond** [comment]
> Please check the PR #36 and confirm if the readme is clear enough regarding the point 3 and 4

I looked at the PR, and I'm more confused:
 - It refers to  a files and directories that do not exist (BBManager/project/ProjectEditor.pro)
 - Still refers to beatbuddy-manager-lite.pro

The comments above are probably factual errors, not something that is correct but needs to be reworded.

**Comment by ericraymond** [comment]
I'm **guessing** that testing these instructions in the readme on a Mac has not been done in a while and is way out of date?

**Comment by JulissaDantes** [comment]
You guessed correctly. When you said "Still refers to beatbuddy-manager-lite.pro" is only referring to it to say what it is, but specifies to use the bbmanagerlean.pro to run the application. 

Note: if you try to build the correct .pro and you get a lot of warning don't worry, the project have warnings that are still being worked on, but if you encounter errors when building that means there is a problem with your kit selection, you might want to check the kit configuration.

**Comment by ericraymond** [comment]
@JulissaDantes I still get the error in #2 from the initial report above.

Are there build instructions to select a kit and/or check the kit configuration?

I suspect anyone who tries to setup and build this project on a Mac will be blocked.....

**Comment by JulissaDantes** [comment]
For the moment, the kit information available is for windows, but you can try to check that out, maybe it can give you some guidance, and possibly you could help us with the Mac building process documentation on the README.md.

**Comment by ericraymond** [comment]
Building a version that runs on the Mac is not supported?

**Comment by ericraymond** [comment]
Still no joy?  Nobody at Singular Sound can provide steps to setup and build that work reliably?

**Comment by JulissaDantes** [comment]
Hi Eric as in right now we are focused on some business priorities, but as soon as we finish with the current work in progress we will review and update the Mac building process.

**Comment by lardoon** [comment]
I don't know if this helps or hinders but I got the mac version to build https://github.com/lardoon/openbbm/runs/891535478?check_suite_focus=true by commenting out the icon lines from the .pro file

```
   ICON_FILES.files += ./mac_deployment/Bundle_src/Content/Resources/Application_Icon.icns
   ICON_FILES.files += ./mac_deployment/Bundle_src/Content/Resources/BeatBuddy_Project_Folder.icns
   ICON_FILES.files += ./mac_deployment/Bundle_src/Content/Resources/BBP.icns
   ICON_FILES.files += ./mac_deployment/Bundle_src/Content/Resources/BBZ.icns
   ICON_FILES.files += ./mac_deployment/Bundle_src/Content/Resources/DRM.icns
   ICON_FILES.files += ./mac_deployment/Bundle_src/Content/Resources/BBS.icns
   ICON_FILES.path  += Contents/Resources
   QMAKE_BUNDLE_DATA += ICON_FILES
   QMAKE_INFO_PLIST   = ./mac_deployment/Bundle_src/Content/Info.plist`
```

I don't have a mac so I can't test the build output

**Comment by iajrz** [comment]
Hey!
So @JustinAiken I just got this to work; issue is latest qt requires that QPainterPath be included directly; adding a
`#include <QPainterPath>`
to the #include section of the file fixes it and lets it run.

**Comment by iajrz** [comment]
> I don't know if this helps or hinders but I got the mac version to build https://github.com/lardoon/openbbm/runs/891535478?check_suite_focus=true by commenting out the icon lines from the .pro file
> 
> ```
>    ICON_FILES.files += ./mac_deployment/Bundle_src/Content/Resources/Application_Icon.icns
>    ICON_FILES.files += ./mac_deployment/Bundle_src/Content/Resources/BeatBuddy_Project_Folder.icns
>    ICON_FILES.files += ./mac_deployment/Bundle_src/Content/Resources/BBP.icns
>    ICON_FILES.files += ./mac_deployment/Bundle_src/Content/Resources/BBZ.icns
>    ICON_FILES.files += ./mac_deployment/Bundle_src/Content/Resources/DRM.icns
>    ICON_FILES.files += ./mac_deployment/Bundle_src/Content/Resources/BBS.icns
>    ICON_FILES.path  += Contents/Resources
>    QMAKE_BUNDLE_DATA += ICON_FILES
>    QMAKE_INFO_PLIST   = ./mac_deployment/Bundle_src/Content/Info.plist`
> ```
> 
> I don't have a mac so I can't test the build output

This works -- at some point we removed those files that are used when building a release version of the software. Removing those lines removes the dependency on those now-missing files.

**Comment by iajrz** [comment]
I am now addressing this issue - first making the changes needed for everything to build properly, then updating the docs. Thank you all for dedicating time to the project.

**Comment by iajrz** [comment]
OK, you should be able to check out from this branch and build straight away -- with the latest Qt available from homebrew.

https://github.com/SingularSound/openbbm/compare/issue34

**Comment by JustinAiken** [comment]
I have qt 5.15.1 installed via homebrew...
MacOS 10.15.6

```bash
$ git checkout issue34
$ cd BBManagerLean && qmake && make clean && make all
.... lots ....
ld: warning: dylib (/usr/local/Cellar/qt/5.15.1/lib/QtNetwork.framework/QtNetwork) was built for newer macOS version (10.13) than being linked (10.12)ld: warning: dylib (/usr/local/Cellar/qt/5.15.1/lib/QtGui.framework/QtGui) was built for newer macOS version (10.13) than being linked (10.12)
ld: warning:
dylib (/usr/local/Cellar/qt/5.15.1/lib/QtMultimedia.framework/QtMultimedia) was built for newer macOS version (10.13) than being linked (10.12)
ld: warning: dylib (/usr/local/Cellar/qt/5.15.1/lib/QtWidgets.framework/QtWidgets) was built for newer macOS version (10.13) than being linked (10.12)
ld: warning: dylib (/usr/local/Cellar/qt/5.15.1/lib/QtCore.framework/QtCore) was built for newer macOS version (10.13) than being linked (10.12)

$ open BBManagerLean.app
LSOpenURLsWithRole() failed with error -10810 for the file /Users/jaiken/projects/openbbm/BBManagerLean/BBManagerLean.app
```

**Comment by iajrz** [comment]
You may want to place the libraries for quazip and minIni in `~/lib/`.

But that's part of the documentation issue -- being addressed in #32 and branch issue32 (which isn't ready, yet) -- the good news is this is building for you, so this particular issue will be closed when I merge the branch.

Please be on the lookout for the documentation fix :)
